### PR TITLE
fix: typewriter shake after press enter

### LINKED ISSUE #520: Typewriter Mode 回车时滚动条大幅跳动影响写作体验
### Description

Typewriter Mode 下，在已经输入了超过一屏文字时回车换行时，滚动条老是跑到上面去了，窗口模式看不到正在写的那一行，全屏时文字内容老是上下跳动。非常影响写作体验。

- [x] Can you reproduce the issue? 

### Steps to reproduce

1. 书写
2. 回车

### Versions

- Mark Text:v 0.13.50
- Operating system:macOS 10.14.1 Beta (18B67a)

### PR DESCRIPTION
**Pull request by MrHeer**
| Q                | A
| ---------------- | ---
| Bug fix?         | yes
| New feature?     | no
| BC breaks?       | no
| Deprecations?    | no
| New tests added? | not needed
| Fixed tickets    | Fixes #520 
| License          | MIT

### Description

the y is 0 after press enter

### CODE REVIEW

File: src/muya/lib/selection/index.js [hunk 1]
```diff
@@ -619,9 +619,13 @@ class Selection {
...
         let rects = range.getClientRects()
         if (rects.length === 0) {
-          rects = range.startContainer && range.startContainer.nodeType === Node.ELEMENT_NODE
-            ? range.startContainer.getClientRects()
+          rects = range.startContainer ? range.startContainer.parentElement.getClientRects()
             : []
+          // prevent tiny vibrations
+          if (rects.length) {
+            const rect = rects[0]
+            rect.y = rect.y + 1
```
[the remark this answers is not in the document]
Comment (MrHeer) [reply src/muya/lib/selection/index.js]: I don't know, I came out through experiments.  After a line break, the cursor moves to the center, but there will be a 1-pixel vibration when entering, the specific reason is unknown.

**Comment by monik81** [comment]
🌙

**Comment by monik81** [comment]
🌘

File: src/muya/lib/selection/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (xinpingwang) [reply src/muya/lib/selection/index.js:627]: I tried this solutions, it works fine. I have reproduce the 1px vibration, seems that only there are content after current editing line will cause this problem. Also, I debug to see the difference between empty line and after content entered. there is actually 1px difference.

left: empty line right: line with content
image

[the remark this answers is not in the document]
Comment (xinpingwang) [reply src/muya/lib/selection/index.js:627]: I tried this solution, it works fine. I also reproduced the 1px vibration (if not plus 1), seem that only there are content after the current editing line will see this problem.  I debug the program, find that there is 1px difference between empty line and after content entered, but still do not why.  

Left: empty line, Right: line with content
image

[the remark this answers is not in the document]
Comment (MrHeer) [reply src/muya/lib/selection/index.js:627]: I agree that there is 1px difference between empty line and after content entered.

[the remark this answers is not in the document]
Comment (fxha) [reply src/muya/lib/selection/index.js:627]: @MrHeer @xinpingwang  Does the fix still work?

[the remark this answers is not in the document]
Comment (MrHeer) [reply src/muya/lib/selection/index.js:627]: I can't install it right now, so I can't test it.

File: src/muya/lib/selection/index.js [hunk 2]
```diff
@@ -619,9 +619,13 @@ class Selection {
         // range.collapse(true)
         let rects = range.getClientRects()
         if (rects.length === 0) {
-          rects = range.startContainer && range.startContainer.nodeType === Node.ELEMENT_NODE
-            ? range.startContainer.getClientRects()
+          rects = range.startContainer ? range.startContainer.parentElement.getClientRects()
```
Comment (fxha) [inline src/muya/lib/selection/index.js:622]: @MrHeer Thanks, the fix working fine and I hope you could add a check for `ELEMENT_NODE` and `TEXT_NODE` like:

```js
          rects = []
          if (range.startContainer && (range.startContainer.nodeType === Node.ELEMENT_NODE || range.startContainer.nodeType === Node.TEXT_NODE)) {
            rects = range.startContainer.parentElement.getClientRects()
          }

          // Prevent tiny vibrations
```

Comment (MrHeer) [reply src/muya/lib/selection/index.js]: @fxha I had updated. Can you test it?

**Review by fxha — APPROVED** [review]

**Pull request merged**